### PR TITLE
GitHub: Retry integration tests when failed

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Resolved / Related Issues

- *Planned internally*

The integration tests became very moody and fail frequently. Let's make it run 3 times at maximum when failed.

### Steps used to test these changes

_None_

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -236,11 +236,16 @@ jobs:
...
-    - name: Run interaction tests
-      run: |
-        dotnet test `
-          $env:AUTOMATED_TESTS_ASSEMBLY_DIR\**\Files.InteractionTests.dll `
-          --logger "trx;LogFileName=$env:AUTOMATED_TESTS_ASSEMBLY_DIR\testResults.trx"
+    # Integration tests became moody lately. Retry 3 times.
+    - uses: nick-fields/retry@v3
+      with:
+        timeout_minutes: 15
+        max_attempts: 3
```
Comment (yaira2) [inline .github/workflows/ci.yml:243]:
```suggestion
        max_attempts: 2
```
Do we need a timeout?

Comment (0x5bfa) [reply .github/workflows/ci.yml]: Looks like the first commit actually failed once. Then this tried again.
Validated ✅

Comment (0x5bfa) [reply .github/workflows/ci.yml]: Looks like timeout field is required.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -236,11 +236,17 @@ jobs:
       shell: pwsh
       run: Start-Process -FilePath "$env:WINAPPDRIVER_EXE86_PATH"
 
+    # Integration tests became moody lately. Retry 3 times.
```
Comment (yaira2) [inline .github/workflows/ci.yml:239]:
```suggestion
    # Retry integration tests if first attempt fails```

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**